File: voom-fe/src/app/shared/rest/vehicle/vehicle.api.ts
import {Api} from '../api';
import {ApiClient} from '../api-client';
import {RequestConfig} from '../rest.model';
import {CreateVehicleDto, VehicleDto} from './vehicle.model';

export default class VehicleApi extends Api {

  constructor(apiClient: ApiClient) {
    super(apiClient);
  }

  getVehicles() {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.get<void, VehicleDto[]>('/api/vehicles', config);
  }

  getVehicle(id: number) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.get<void, VehicleDto>(`/api/vehicles/${id}`, config);
  }

  createVehicle(vehicleDto: CreateVehicleDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.post<CreateVehicleDto, VehicleDto>('/api/vehicles', vehicleDto, config);
  }

  updateVehicle(id: number, vehicleDto: VehicleDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.put<VehicleDto, VehicleDto>(`/api/vehicles/${id}`, vehicleDto, config);
  }

  deleteVehicle(id: number) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.delete<void, void>(`/api/vehicles/${id}`, config);
  }


}

File: voom-app/app/src/main/java/inc/visor/voom/app/shared/api/VehicleApi.java
package inc.visor.voom.app.shared.api;

import java.util.List;

import inc.visor.voom.app.shared.dto.vehicle.CreateVehicleDto;
import inc.visor.voom.app.shared.dto.vehicle.VehicleDto;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface VehicleApi {

    @GET("api/vehicles")
    Call<List<VehicleDto>> getVehicles();

    @GET("api/vehicles/{id}")
    Call<VehicleDto> getVehicle(@Path("id") long id);

    @POST("api/vehicles")
    Call<VehicleDto> createVehicle(@Body CreateVehicleDto dto);

    @PUT("api/vehicles/{id}")
    Call<VehicleDto> updateVehicle(@Path("id") long id, @Body VehicleDto dto);

    @DELETE("api/vehicles/{id}")
    Call<Void> deleteVehicle(@Path("id") long id);

}
